packages/ui/react-ui-stack/src/components/Section.tsx

```diff
@@ -2,48 +2,48 @@
 // Copyright 2023 DXOS.org
 //
 
-import { DotsSixVertical, X, ArrowSquareOut, DotsThreeVertical } from '@phosphor-icons/react';
+import { useFocusableGroup, useTabsterAttributes } from '@fluentui/react-tabster';
+import { ArrowSquareOut, CaretUpDown, DotsNine, type IconProps, X } from '@phosphor-icons/react';
+import * as CollapsiblePrimitive from '@radix-ui/react-collapsible';
 import React, {
   forwardRef,
   useState,
   type ForwardRefExoticComponent,
   type RefAttributes,
   type FC,
   type PropsWithChildren,
+  type Dispatch,
+  type SetStateAction,
 } from 'react';
 
-import { Button, DensityProvider, DropdownMenu, List, ListItem, useTranslation } from '@dxos/react-ui';
+import { Button, DropdownMenu, List, ListItem, useTranslation, type TFunction } from '@dxos/react-ui';
 import {
   type MosaicActiveType,
   type MosaicDataItem,
   type MosaicTileComponent,
   type MosaicTileProps,
   useMosaic,
 } from '@dxos/react-ui-mosaic';
-import {
-  fineButtonDimensions,
-  focusRing,
-  getSize,
-  hoverableControlItem,
-  hoverableControls,
-  hoverableFocusedControls,
-  hoverableFocusedKeyboardControls,
-  hoverableOpenControlItem,
-  attentionSurface,
-  mx,
-  staticFocusRing,
-  staticHoverableControls,
-} from '@dxos/react-ui-theme';
+import { focusRing, getSize, mx } from '@dxos/react-ui-theme';
 
+import { CaretDownUp } from './CaretDownUp';
+import { DropDownMenuDragHandleTrigger } from './DropDownMenuDragHandleTrigger';
+import { stackColumns } from './style-fragments';
 import { translationKey } from '../translations';
 
+const sectionActionDimensions = 'p-1 shrink-0 min-bs-0 is-[--rail-action] bs-min';
+
 export type StackSectionContent = MosaicDataItem & { title?: string };
 
+export type CollapsedSections = Record<string, boolean>;
+
 export type StackContextValue<TData extends StackSectionContent = StackSectionContent> = {
   SectionContent: FC<{ data: TData }>;
   transform?: (item: MosaicDataItem, type?: string) => StackSectionItem;
   onDeleteSection?: (path: string) => void;
   onNavigateToSection?: (id: string) => void;
+  collapsedSections?: CollapsedSections;
+  setCollapsedSections?: Dispatch<SetStateAction<CollapsedSections>>;
```

Comment:
`onCollapseSection?: (id: string) => void`?

Reply:
I want to make a distinction between two patterns:

- `{prop, onChangeProp, defaultProp}` when controllable
- `{prop, setProp}` when drilling `useState` down

For me, the name `setCollapsedSections` signals that this will directly affect a state, as much as I’d like to use something that sounds more natural like you suggest.

Do you think we should make exceptions for contexts, or is your suggestion applying a principle we can apply more broadly?

Reply:
I'm curious why you want to make that distinction? My sense would be that the component shouldn't care what the callback is being mapped back to.

All that aside, I think it mapping directly back to `useState` is going to be a temporary state based on this TODO https://github.com/dxos/dxos/pull/5917#discussion_r1530526158. Which another reason why letting how it's being used influence the api seems like a bad idea to me.

Reply:
I see, so in other words the principle to apply is “name it something intentionally different because its implementation should not make any guarantees”? I think I’m amenable to that.

Reply:
It would be similar to `handle…` vs `on…`, where the former is only used locally. Essentially we argue `set…` should be local-only as well.

Reply:
Ah, however, I notice your suggestion reframes this as a singular. Because the implementation of this state should change, I intentionally didn’t try to split the collective state setter into single state setters for each section, to avoid complexity we’re soon to remove.

For now I think it’s best to just leave a TODO here.

Reply:
`onCollapseSection?: (sections: CollapsedSections) => void` with a todo explaining why it's passing all of them then?

Reply:
I suppose either leaving the code alone and adding a comment about it, or making changes to the code in addition to adding a comment about it, are roughly equivalent approaches to me. I’m also concerned about minting `onCollapseSection` when that might come off as a design assertion when we might refactor this to use a pattern e.g. signals which doesn’t have those callbacks.

Reply:
Yeah, I think I'm trying to make a design assertion on the api. @zhenyasav and I discussed the reactive object pattern and the conclusion was basically, it's an optimization and you still need callbacks. The reason being that you need somewhere to inject business logic, so the component can't be directly mutating the props. There may be exceptions and we could consider defaults for the callbacks which are simple mutation pass-throughs but we still need the callbacks in most scenarios.

Reply:
I see, got it, will make the change in that case.